packages/kbn-reporting/export_types/csv/csv_v2.ts

```diff
@@ -121,6 +117,36 @@ export class CsvV2ExportType extends ExportType<
 
     // use Discover contract to convert the job params into inputs for CsvGenerator
     const locatorClient = await discoverPluginStart.locator.asScopedClient(fakeRequest);
+
+    const query = await locatorClient.queryFromLocator(params);
+
+    if (query && 'esql' in query) {
+      // TODO: use columnsFromLocator
+      // We don't implement it now because it might be a breaking change for csv_v2 api that for non esql only works with saved search
+      // https://github.com/elastic/kibana/issues/151190
+      // const columns = await locatorClient.columnsFromLocator(params);
```

Comment:
I think I don't follow this. Is it saying there would need to be changes made to the `columnsFromLocator` method in order to have it support esql, and those changes could be breaking for non-esql?

Reply:
> Is it saying there would need to be changes made to the columnsFromLocator method in order to have it support esql, and those changes could be breaking for non-esql?

Yes. I'll try to make the comment clear.


currently locatorClient.columnsFromLocator can only extract columns from the saved search, but for the es|ql I simply wanted to get currently visible columns from params.columns. 
I didn't want to add this change inside locatorClient.columnsFromLocator, as it would change the behaviour of csv_v2 for non-ES|QL export